File: easy/1903_largest_odd_number_in_string/Solution.java
class Solution {
  public String largestOddNumber(String num) {
    int lastOdd = 0;

    for (int i = 0; i < num.length(); ++i) {
      if ((num.charAt(i) - 48) % 2 == 1) {
        lastOdd = i + 1;
      }
    }

    return num.substring(0, lastOdd);
  }
}

File: easy/1903_largest_odd_number_in_string/solution.cpp
#include <string>

class Solution {
public:
  std::string largestOddNumber(std::string num) {
    size_t lastOdd = 0;

    for (size_t i = 0; i < num.size(); ++i) {
      if ((num[i] - 48) % 2 == 1)
        lastOdd = i + 1;
    }

    return num.substr(0, lastOdd);
  }
};
